### DISCUSSION TITLE
Crash at evbuffer_get_length

### DISCUSSION THREAD

**Discussion by dantearmok** [transmission/transmission#5411] [Support]
Mac mini (late 2012, 2.5-GHz 4-Core i5), 16GB RAM, plus an assortment of USB & FW HDs.
macOS 10.15 Catalina, ClamXAV, PeakHour, PIA, and Xcode 12.3 (all fully updated).

Running Transmission-d4e500245b Build #9021 (Apr 17, 2023, 7:20:23 PM)

200+ active torrents, maybe a dozen leeching.  After a few minutes, Transmission crashes.  This is repeatable.

Process:               Transmission [9093]
Path:                  /Applications/Transmission.app/Contents/MacOS/Transmission
Identifier:            Transmission
Version:               4.1.0-dev (14714.4.1.0)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Transmission [9093]
User ID:               502

Date/Time:             2023-04-17 22:59:04.413 -0400
OS Version:            Mac OS X 10.15.7 (19H2026)
Report Version:        12
Anonymous UUID:        9D83084F-22F2-20FB-B0D2-056E94905F2F

Time Awake Since Boot: 2700 seconds

System Integrity Protection: enabled

Crashed Thread:        1

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000700000036
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [9093]

VM Regions Near 0x700000036:
    CoreAnimation          0000000111fd1000-00000001121f2000 [ 2180K] rw-/rwx SM=PRV  
--> 
    MALLOC_NANO            0000600000000000-0000600008000000 [128.0M] rw-/rwx SM=PRV  

Thread 0:: Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	0x00007fff6be9ddea mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff6be9e160 mach_msg + 60
2   com.apple.CoreFoundation      	0x00007fff31fdbd07 __CFRunLoopServiceMachPort + 247
3   com.apple.CoreFoundation      	0x00007fff31fda7c1 __CFRunLoopRun + 1329
4   com.apple.CoreFoundation      	0x00007fff31fd9c33 CFRunLoopRunSpecific + 466
5   com.apple.HIToolbox           	0x00007fff30bf5aad RunCurrentEventLoopInMode + 292
6   com.apple.HIToolbox           	0x00007fff30bf57c5 ReceiveNextEventCommon + 584
7   com.apple.HIToolbox           	0x00007fff30bf5569 _BlockUntilNextEventMatchingListInModeWithFilter + 64
8   com.apple.AppKit              	0x00007fff2f23e3c9 _DPSNextEvent + 883
9   com.apple.AppKit              	0x00007fff2f23cc10 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1352
10  com.apple.AppKit              	0x00007fff2f22e91e -[NSApplication run] + 658
11  com.apple.AppKit              	0x00007fff2f200706 NSApplicationMain + 777
12  libdyld.dylib                 	0x00007fff6bd5ccc9 start + 1

Thread 1 Crashed:
0   org.m0k.transmission          	0x000000010d324ef3 evbuffer_get_length + 13
1   org.m0k.transmission          	0x000000010d2b9b0e libtransmission::Buffer::clear() + 18
2   org.m0k.transmission          	0x000000010d2b0064 (anonymous namespace)::canRead(tr_peerIo*, void*, unsigned long*) + 1353
3   org.m0k.transmission          	0x000000010d29aa1e tr_peerIo::can_read_wrapper() + 164
4   org.m0k.transmission          	0x000000010d29acc4 tr_peerIo::try_read(unsigned long) + 292
5   org.m0k.transmission          	0x000000010d2d066a tr_bandwidth::phaseOne(std::__1::vector<tr_peerIo*, std::__1::allocator<tr_peerIo*> >&, tr_direction) + 544
6   org.m0k.transmission          	0x000000010d2d0caa tr_bandwidth::allocate(unsigned int) + 846
7   org.m0k.transmission          	0x000000010d2a35cc tr_peerMgr::bandwidthPulse() + 132
8   org.m0k.transmission          	0x000000010d332e4b event_process_active_single_queue + 1121
9   org.m0k.transmission          	0x000000010d32fceb event_base_loop + 1134
10  org.m0k.transmission          	0x000000010d28f035 tr_session_thread_impl::sessionThreadFunc(event_base*) + 87
11  org.m0k.transmission          	0x000000010d28f427 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (tr_session_thread_impl::*)(event_base*), tr_session_thread_impl*, event_base*> >(void*) + 63
12  libsystem_pthread.dylib       	0x00007fff6bf65109 _pthread_start + 148
13  libsystem_pthread.dylib       	0x00007fff6bf60b8b thread_start + 15

Thread 2:
0   libsystem_kernel.dylib        	0x00007fff6bea60ee __select + 10
1   libcurl.4.dylib               	0x00007fff699feb5b Curl_poll + 623
2   libcurl.4.dylib               	0x00007fff699f9af1 Curl_multi_wait + 609
3   org.m0k.transmission          	0x000000010d2cba1e tr_web::Impl::curlThreadFunc() + 414
4   org.m0k.transmission          	0x000000010d2cbcb4 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (tr_web::Impl::*)(), tr_web::Impl*> >(void*) + 59
5   libsystem_pthread.dylib       	0x00007fff6bf65109 _pthread_start + 148
6   libsystem_pthread.dylib       	0x00007fff6bf60b8b thread_start + 15

Thread 3:
0   libsystem_kernel.dylib        	0x00007fff6bea2756 kevent + 10
1   org.m0k.transmission          	0x000000010d23de7c -[VDKQueue watcherThread:] + 203
2   com.apple.Foundation          	0x00007fff34695222 __NSThread__start__ + 1064
3   libsystem_pthread.dylib       	0x00007fff6bf65109 _pthread_start + 148
4   libsystem_pthread.dylib       	0x00007fff6bf60b8b thread_start + 15

Thread 4:
0   libsystem_pthread.dylib       	0x00007fff6bf60b68 start_wqthread + 0

Thread 5:
0   libsystem_pthread.dylib       	0x00007fff6bf60b68 start_wqthread + 0

Thread 6:
0   libsystem_pthread.dylib       	0x00007fff6bf60b68 start_wqthread + 0

Thread 7:
0   libsystem_pthread.dylib       	0x00007fff6bf60b68 start_wqthread + 0

Thread 8:
0   libsystem_pthread.dylib       	0x00007fff6bf60b68 start_wqthread + 0

Thread 9:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	0x00007fff6be9ddea mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff6be9e160 mach_msg + 60
2   com.apple.CoreFoundation      	0x00007fff31fdbd07 __CFRunLoopServiceMachPort + 247
3   com.apple.CoreFoundation      	0x00007fff31fda7c1 __CFRunLoopRun + 1329
4   com.apple.CoreFoundation      	0x00007fff31fd9c33 CFRunLoopRunSpecific + 466
5   com.apple.AppKit              	0x00007fff2f3dfbf4 _NSEventThread + 132
6   libsystem_pthread.dylib       	0x00007fff6bf65109 _pthread_start + 148
7   libsystem_pthread.dylib       	0x00007fff6bf60b8b thread_start + 15

Thread 1 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000700000006  rcx: 0x0000000000000050  rdx: 0x0000000000001d45
  rdi: 0x0000000700000006  rsi: 0x00000000000320a8  rbp: 0x000070000b8073a0  rsp: 0x000070000b807380
   r8: 0x00000000000320a8   r9: 0x0000000000000680  r10: 0x00007f871c06cd10  r11: 0x00007f871c06cd08
  r12: 0x00007f871db5d600  r13: 0x0000000000000000  r14: 0x00007f871db5e0c8  r15: 0x0000000000001d45
  rip: 0x000000010d324ef3  rfl: 0x0000000000010286  cr2: 0x0000000700000036
  
Logical CPU:     0
Error Code:      0x00000004 (no mapping for user data read)
Trap Number:     14

**Comment by dantearmok** [comment]
2 more crashes today with the same-looking crash.

**Comment by ckerr** [comment]
Converted to issue @ https://github.com/transmission/transmission/issues/5422
